src/main/java/de/tum/cit/aet/artemis/iris/domain/settings/IrisCourseSettings.java

```diff
@@ -52,6 +52,10 @@ public class IrisCourseSettings extends IrisSettings {
     @JoinColumn(name = "iris_competency_generation_settings_id")
     private IrisCompetencyGenerationSubSettings irisCompetencyGenerationSettings;
 
+    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
```

Comment:
The many eager fetches here can become a problem with Hibernate. We definitely need to change this in the future, this has become way too complex when you think about that a few boolean values are stored.

Reply:
@Hialus I suggest we talk about this asap